### PR TITLE
Added practice feature and fixed line going out of the progressbar

### PR DESCRIPTION
**Pull request by Termantita**
## Related project links
- [Conditional checkpoint sprite](https://github.com/users/dankmeme01/projects/1?pane=issue&itemId=64206577)
- [Line glitch](https://github.com/users/dankmeme01/projects/1?pane=issue&itemId=65262120)

## Demostration
image

Was tested in the main branch with normal servers (protocol-v5).

### CODE REVIEW

File: src/ui/game/progress/progress_icon.cpp [hunk 1]
```diff
@@ -14,18 +15,26 @@ bool PlayerProgressIcon::init() {
...
 
     auto& settings = GlobedSettings::get();
-
     auto gm = GameManager::get();
     auto color1 = gm->colorForIdx(data.color1);
     auto color2 = gm->colorForIdx(data.color2);
 
     Build<CCLayerColor>::create(ccc4(color1.r, color1.g, color1.b, 255), 2.f, 8.f)
-        .pos(0.f, 3.f)
+        .pos(0.f, 4.f)
```
Comment (Termantita) [inline src/ui/game/progress/progress_icon.cpp:27]: it wassnt centered

File: src/ui/game/progress/progress_icon.cpp [hunk 2]
```diff
@@ -36,14 +45,19 @@ void PlayerProgressIcon::updateIcons(const PlayerIconData& data) {
...
-void PlayerProgressIcon::updatePosition(float progress) {
+void PlayerProgressIcon::updatePosition(float progress, bool isPracticing) {
     auto parent = this->getParent()->getParent();
     if (!parent) return;
 
     CCSize pbSize = parent->getScaledContentSize();
     float prOffset = (pbSize.width - 2.f) * progress;
 
-    this->toggleLine(progress > 0.01f && progress < 0.99f);
+    bool lineIsVisible = progress > 0.01f && progress < 0.99f;
```
Comment (Termantita) [inline src/ui/game/progress/progress_icon.cpp:55]: Maybe the upper value is too high for the practice sprite, looks kinda weird

File: src/ui/game/progress/progress_icon.cpp [hunk 3]
```diff
@@ -14,18 +15,26 @@ bool PlayerProgressIcon::init() {
...
-
     auto gm = GameManager::get();
     auto color1 = gm->colorForIdx(data.color1);
     auto color2 = gm->colorForIdx(data.color2);
 
     Build<CCLayerColor>::create(ccc4(color1.r, color1.g, color1.b, 255), 2.f, 8.f)
-        .pos(0.f, 3.f)
+        .pos(0.f, 4.f)
         .parent(this)
+        .scaleY(.8f)
```
Comment (Termantita) [inline src/ui/game/progress/progress_icon.cpp:29]: Fixes line traspassing the progressbar

File: src/ui/game/progress/progress_icon.cpp [hunk 4]
```diff
@@ -1,4 +1,5 @@
 #include "progress_icon.hpp"
+#include "data/types/gd.hpp"
```
Comment (dankmeme01) [inline src/ui/game/progress/progress_icon.cpp:2]: unneeded include

**Review by dankmeme01 — CHANGES_REQUESTED** [review]

**Pull request merged**